### PR TITLE
Remove b and casino support from convert4qmc

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

Remove these features. We have no inputs or knowledge of the "b" code used for AGP. The casino support for single determinant gaussian wavefunctions is not perfectly working, untested, and there are alternative routes to handle this case. e.g. molden or PySCF converter.

## What type(s) of changes does this code introduce?

- Feature removal
- Build related changes
- Documentation changes

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?

nitrogen, llvm.

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- NA. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by ye-luo — DISMISSED** [review]
The BMakeFunc seems intended for TurboRVB.
All the removed code can be resurrected from history if a need shows up.

**Comment by prckent** [comment]
Good point Ye. These would be trivial to add back. The converter is nicely modular.

I foresee a similar discussion around the AGP wavefunction support in QMCPACK.

**Comment by prckent** [comment]
Some strange results from codecov/patch, perhaps because files were removed.

As an experiment I'll add a test to catch the untested but clearly not critical convert4qmc usage lines.

**Comment by prckent** [comment]
(OK to review/merge. Large apparent diff in last update due to clang-format and indentation of try {} block. )

File: src/QMCTools/tests/CMakeLists.txt [hunk 1]
```diff
@@ -70,4 +70,10 @@ ELSE()
...
 SET_TESTS_PROPERTIES(extract-eshdf-kvectors_2x1x1 PROPERTIES TIMEOUT 120 LABELS "deterministic" PASS_REGULAR_EXPRESSION "0   0   0\n0.5   0   0")
 
-
+# Minimal test for convert4qmc. Check for Usage on output if no arguments. Primarily tested through converter tests.
+IF ( HAVE_MPI )
+  ADD_TEST( NAME convert4qmc_usage COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} 1 ${MPIEXEC_PREFLAGS} ${qmcpack_BINARY_DIR}/bin/convert4qmc )
+ELSE()
+  ADD_TEST(NAME convert4qmc_usage COMMAND "${qmcpack_BINARY_DIR}/bin/convert4qmc")
+ENDIF()
+SET_TESTS_PROPERTIES(convert4qmc_usage PROPERTIES TIMEOUT 120 LABELS "deterministic" PASS_REGULAR_EXPRESSION "Usage:")
```
Comment (ye-luo) [inline src/QMCTools/tests/CMakeLists.txt:79]: add_unit_test macro as
```
                              PROCESSORS ${TOT_PROCS}
                              ENVIRONMENT OMP_NUM_THREADS=${THREADS}
                              PROCESSOR_AFFINITY TRUE )
```
So it is better to be consistent with other ones.
I think OMP_NUM_THREADS can be just pick magic number 3 if you are not sure whether a test uses OpenMP.
So please add these test properties to this one as well qmc-ckeck-affinity and the one handling supercell (probably all tests recently added) in a separate PR.

**Review by ye-luo — APPROVED** [review]
LGTM

File: src/QMCTools/tests/CMakeLists.txt (hunk 1 above)
Comment (prckent) [reply src/QMCTools/tests/CMakeLists.txt:79]: Incoming

Comment (prckent) [reply src/QMCTools/tests/CMakeLists.txt]: Pushed

**Review by ye-luo — APPROVED** [review]

**Pull request merged**